### PR TITLE
Systemd delta tweaks

### PR DESCRIPTION
**Pull request by fbuihuu**

### CODE REVIEW

File: src/delta/delta.c [hunk 1]
```diff
@@ -392,19 +392,28 @@ static int enumerate_dir(
...
-        r = chase_symlinks(p, NULL, 0, &target);
-        if (r < 0)
-                return r;
+        dirname = strjoina(prefix, "/", suffix);
 
-        return !streq(p, target) && nulstr_contains(prefixes, target);
-#else
-        return 0;
+        if (chase_symlinks(dirname, NULL, 0, &target) < 0)
+                return false;
```
Comment (keszybz) [inline src/delta/delta.c:404]: Before, we could fail e.g. with EPERM. Now, such errors will be silently skipped. I don't think this change is warranted.

Comment (fbuihuu) [reply src/delta/delta.c:404]: We defer the error handling to the next caller which is going to fail anyways, Here we're only interested to see what is the target of the symlink but not in following it.

**Comment by fbuihuu** [comment]
I'll see if I can address your points.

**Comment by fbuihuu** [comment]
> This line is nonsensical. First of all, the docs say that "If [prefix] is given, only overriding files contained in this directory will be shown". It is unclear what "overriding file" means in case of symlinks: the symlink or the target?

I think it's confusing mostly due to the wording: "overriding file" should be read "overridden file".  In this case it becomes pretty clear (at least from my point of view) that the target is concerned or IOW `systemd-delta /usr/lib` shows overrides that could come from any upper "layers" that is: /run, /etc.

In its turn `systemd-delta /etc` shouldn't show any overrides since there's no upper layer above /etc.

I'm really not sure about your suggested change, i.e. to look at the symlink: in this case one couldn't find easily the overrides of a given configuration file anymore . For instance to see the customizations done over the distros config files one should do: `for p in /run /etc; do systemd-delta $p; done`.

> Second, /etc/systemd/system/dbus-org.freedesktop.timedate1.service is actually a symlink to /usr/lib/systemd/system/timedatex.service. But our output format strongly suggests that it is a symlink to /usr/lib/systemd/system/dbus-org.freedesktop.timedate1.service.

I agree here the output of systemd-delta is quite misleading but I must admit that I currently can't see how we could improve it properly.

Maybe for overridden or masked configuration files we could use something like:

```
$ systemd-delta /usr/lib
[MASKED]    /usr/lib/systemd/system/tmp.mount ← /etc
[OVERRIDEN] /usr/lib/udev/rules.d/60-serial.rules  ← /etc
```
IOW we start by showing the overridden files and show only the prefix where the overrides live ? In this case there's no point to show the suffix and the filename of the overrides since they're the same as the overridden ones. Here the output shouldn't suggest the use of any symlink.

Regarding the example with `[REDIRECT]` you gave  maybe we could use a similar output as the one used when overrides have the same content as the original files.

So similarly to:

```
[OVERRIDDEN] /etc/udev/rules.d/60-serial.rules → /usr/lib/udev/rules.d/60-serial.rules

Files /usr/lib/udev/rules.d/60-serial.rules and /etc/udev/rules.d/60-serial.rules are identical
```

we could show `[REDIRECT]` like the following:

```
[OVERRIDEN] /usr/lib/systemd/system/default.target <- /etc

File /etc/systemd/system/default.target is a symlink to /usr/lib/systemd/system/multi-user.target
```
WDYT ?

**Comment by fbuihuu** [comment]
@keszybz: while looking at the code I think I found another bug. So I'm adding a new commit on top of the ones I already pushed.

**Comment by fbuihuu** [comment]
@keszybz could you please consider merging the bug fixes in this PR ?

I'm fine to rework the output of delta later but I think we should fix the current code.

**Comment by fbuihuu** [comment]
@keszybz could you please merge the 2 fixes if they're correct ?

Improving the output of systemd-delta should be done in another step anyway.

Thanks !

**Comment by fbuihuu** [comment]
@keszybz, thanks !

**Pull request merged**